### PR TITLE
Applying filtering before directives are applied

### PR DESCRIPTION
**Pull request by nitinmotgi**
This feature adds a pre-condition expression based filter that can be applied. 

precondition

screen shot 2017-03-01 at 12 09 45 am

### CODE REVIEW

File: transform/src/main/java/co/cask/wrangler/PreconditionException.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package co.cask.wrangler;
+
+/**
+ * Thrown when there are issues with pre-conditions.
+ */
+public class PreconditionException extends Exception {
```
Comment (bdmogal) [inline transform/src/main/java/co/cask/wrangler/PreconditionException.java:22]: if this extends IllegalArgumentException instead, will it simplify exception handling?

File: transform/src/test/java/co.cask.wrangler/PreconditionTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package co.cask.wrangler;
```
Comment (bdmogal) [inline transform/src/test/java/co.cask.wrangler/PreconditionTest.java:17]: license

File: transform/src/main/java/co/cask/wrangler/PreconditionException.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package co.cask.wrangler;
```
Comment (bdmogal) [inline transform/src/main/java/co/cask/wrangler/PreconditionException.java:17]: license

**Review by bdmogal — COMMENTED** [review]
1 question abt exception  handling. rest lgtm

File: transform/src/main/java/co/cask/wrangler/PreconditionException.java (hunk 1 above)
Comment (nitinmotgi) [reply transform/src/main/java/co/cask/wrangler/PreconditionException.java:22]: No runtime exception **EVER** also I need that during evaluation too.

File: transform/src/main/java/co/cask/wrangler/PreconditionException.java (hunk 3 above)
Comment (nitinmotgi) [reply transform/src/main/java/co/cask/wrangler/PreconditionException.java:17]: done!

File: transform/src/test/java/co.cask.wrangler/PreconditionTest.java (hunk 2 above)
Comment (nitinmotgi) [reply transform/src/test/java/co.cask.wrangler/PreconditionTest.java:17]: done!

**Review by anwar6953 — APPROVED** [review]

File: transform/src/test/java/co.cask.wrangler/PreconditionTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    Assert.assertEquals(true, new Precondition("true").apply(record));
+    Assert.assertEquals(false, new Precondition("false").apply(record));
+  }
+
+  @Test(expected = PreconditionException.class)
+  public void testBadCondition() throws Exception {
+    Record record = new Record("a", 1).add("b", "x").add("c", 2.06);
+    Assert.assertEquals(true, new Precondition("c").apply(record));
+  }
+}
```
Comment (anwar6953) [inline transform/src/test/java/co.cask.wrangler/PreconditionTest.java]: nit - missing newline at EOF

Comment (nitinmotgi) [reply transform/src/test/java/co.cask.wrangler/PreconditionTest.java]: addressed

**Pull request merged**